storage/cache/subtree_cache.go

```diff
@@ -168,17 +168,9 @@ func (s *SubtreeCache) preload(ids []tree.NodeID, getSubtrees GetSubtreesFunc) e
 		}
 		delete(want, string(t.Prefix))
 	}
-
-	// We might not have got all the subtrees we requested, if they don't already exist.
-	// Create empty subtrees for anything left over. Note that multiple parallel readers
-	// may be be running this code and touch the same keys, although this doesn't happen
-	// normally.
-	for _, id := range want {
-		if err := s.cacheSubtree(s.newEmptySubtree(id)); err != nil {
-			return err
-		}
+	if r := len(want); r != 0 {
```

Comment:
I'm not 100% sure if there's an edge case where this happens. It might possibly be where the tree grows enough to require a new tile that wasn't previously written.

Reply:
The key point is in the PR description: we will not try to read this new tile with `GetMerkleNodes` (and hence `SubtreeCache.preload`). The latter method is used in 2 scenarios:
1. Sequencer's transaction: read compact range of the current tree. These nodes (and hence tiles) always exist, unless someone removed them from the DB.
2. Reading proofs. These "read" transactions also know the current tree size, and request the existing nodes.

Reply:
This is a leftover from maps, I believe. There we could read a leaf which previously did not exist, so we would need empty tiles silently added.

Reply:
Maybe it was added for the map. It's a long time ago now.